### ISSUE TITLE
Statefulness in Text generation using a RNN with eager execution

### ISSUE THREAD

**Issue by ZhiliangWu** [tensorflow/tensorflow#28766]
## URL(s) with the issue:

https://www.tensorflow.org/tutorials/sequences/text_generation

### Clear description

* Two questions regarding the `stateful` defined in the tutorial. 
* The input data was originally a long text. It got cut into sequences with length 100, shuffled, and packed into batches of size 64. However, according to the [documentation](https://www.tensorflow.org/api_docs/python/tf/keras/layers/RNN) for the `statefulness`
> You can set RNN layers to be 'stateful', which means that the states
computed for the samples in one batch will be reused as initial states
for the samples in the next batch. This assumes a one-to-one mapping
between samples in different successive batches.

After pre-processing the data as discussed above, there is no such one-to-one mapping is not there, i.e. the samples in different batches are independent instead of related. This is confirmed by running the following code after the code for create training batches, which print out the successive samples with index `0` in the first two batches. 
```Python
for input_example, target_example in  dataset.take(2):
  print ('Input data: ', repr(''.join(idx2char[input_example[0, :].numpy()])))
```
outputs
```Python
Input data:  'n that perished vessel the dowry of his\nsister. But mark how heavily this befell to the\npoor gentlew'
Input data:  "y enrich'd\nWith politic grave counsel; then the king\nHad virtuous uncles to protect his grace.\n\nFirs"
```
---
* In addition, if `stateful=True`, it is suggested to set `shuffle=False` in the `model.fit()`, which is also missing in the tutorial. 
> specify `shuffle=False` when calling fit().

* The problems above are really confusing and any discussions are welcome.

**Comment by siavash-khodadadeh** [comment]
https://www.tensorflow.org/beta/tutorials/text/text_classification_rnn

The same thing is true about the same tutorial for TF 2.0 and it is very confusing.

**Comment by Saduf2019** [comment]
@ZhiliangWu 
Is this still an issue

**Comment by MarkDaoust** [comment]
No. This is fixed.
